model/patch_lifecycle.go

```diff
@@ -251,51 +226,156 @@ func GetPatchedProject(ctx context.Context, settings *evergreen.Settings, p *pat
 	}
-	var pc *ProjectConfig
+
+	var pc string
 	if projectRef.IsVersionControlEnabled() {
-		pc, err = CreateProjectConfig(projectFileBytes, p.Project)
+		pc, err = getProjectConfigYAML(p, projectFileBytes)
 		if err != nil {
-			return nil, nil, errors.WithStack(err)
+			return nil, nil, errors.Wrap(err, "getting patched project config")
 		}
 	}
+
+	project := &Project{}
+	pp, err := LoadProjectInto(ctx, projectFileBytes, opts, p.Project, project)
+	if err != nil {
+		return nil, nil, errors.WithStack(err)
+	}
+	// LoadProjectInto does not set the parser project's identifier, so set it
+	// here.
+	pp.Identifier = utility.ToStringPtr(p.Project)
 	ppOut, err := yaml.Marshal(pp)
 	if err != nil {
 		return nil, nil, errors.Wrap(err, "marshalling parser project into YAML")
 	}
+
 	patchConfig := &PatchConfig{
-		PatchedParserProject: string(ppOut),
-	}
-	if pc != nil {
-		pcOut, err := yaml.Marshal(pc)
-		if err != nil {
-			return nil, nil, errors.Wrap(err, "marshalling project config into YAML")
-		}
-		patchConfig.PatchedProjectConfig = string(pcOut)
+		PatchedParserProjectYAML: string(ppOut),
+		PatchedParserProject:     pp,
+		PatchedProjectConfig:     pc,
 	}
 	return project, patchConfig, nil
 }
 
-// MakePatchedConfig takes in the path to a remote configuration a stringified version
-// of the current project and returns an unmarshalled version of the project
-// with the patch applied
+// GetPatchedProjectConfig returns the project configuration by fetching the
+// latest commit information from GitHub and applying the patch to the latest
+// remote configuration. The error returned can be a validation error.
+func GetPatchedProjectConfig(ctx context.Context, settings *evergreen.Settings, p *patch.Patch, githubOauthToken string) (string, error) {
+	if p.Version != "" {
+		return "", errors.Errorf("patch '%s' already finalized", p.Version)
+	}
+
+	if p.ProjectStorageMethod != "" || p.PatchedParserProject != "" {
+		// If the patch has been created but not finalized, it has either
+		// already saved the project config as a string (if any), the parser
+		// project document, or has the entire patched parser project as a
+		// string (which is the fallback for old patches that don't store the
+		// parser project). Since the project config is optional, the patch may
+		// be created and have already evaluated the patched project config, but
+		// there simply was none. Therefore, this is a valid way to check and
+		// get the PatchedProjectConfig after the patch is already created.
+		return p.PatchedProjectConfig, nil
+	}
+
+	projectRef, opts, err := getLoadProjectOptsForPatch(p, githubOauthToken)
+	if err != nil {
+		return "", errors.Wrap(err, "fetching project options for patch")
+	}
+
+	projectFileBytes, err := getPatchedProjectYAML(ctx, projectRef, opts, p)
+	if err != nil {
+		return "", errors.Wrap(err, "getting patched project file as YAML")
+	}
+
+	if !projectRef.IsVersionControlEnabled() {
```

Comment:
We can move this above the `getPatchedProjectYAML` call.

Reply:
Nice catch, done.